collector/zfs_linux.go

```diff
@@ -0,0 +1,88 @@
+package collector
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+const (
+	zfsArcstatsProcpath = "spl/kstat/zfs/arcstats"
+)
+
+func (c *zfsCollector) zfsAvailable() (err error) {
+	file, err := c.openArcstatsFile()
+	if err != nil {
+		file.Close()
```

Comment:
And actually you *only* need to close the file when the opening was successful. When there was an error, you won't even get a valid `*File` back, but `nil`. So the code above would actually crash the entire node exporter with a nil pointer dereference panic in case the file can't be read.